AndelaOSP/art-android
# [156718829] Outgoing requests should have a token

**What does this PR do?**
Provides an interceptor so that all outgoing requests can have a token.

**How should this be manually tested?**
- Log in to the app.
- Access an endpoint that requires token authentication, the request should go through.
- Alternatively use the logged out put in log cat and get "AUTH_TOKEN" and verify if the token can 
   access endpoints using postman.

**Relevant PT stories**
https://www.pivotaltracker.com/story/show/156718829
